src/beast/beast/http/src/test/beast_http_rfc2616_test.cpp

```diff
@@ -0,0 +1,128 @@
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <beast/http/rfc2616.h>
+#include <beast/unit_test/suite.h>
+#include <string>
+#include <vector>
+
+
+namespace beast {
+namespace rfc2616 {
+namespace test {
+
+class rfc2616_test : public beast::unit_test::suite
+{
+public:
+    void
+    checkSplit(std::string const& s,
+        std::vector <std::string> const& expected)
+    {
+        auto const parsed = split_commas(s.begin(), s.end());
+        expect (parsed == expected);
+    }
+
+    void testSplit()
+    {
+        checkSplit("",              {});
+        checkSplit(" ",             {});
+        checkSplit("  ",            {});
+        checkSplit("\t",            {});
+        checkSplit(" \t ",          {});
+        checkSplit(",",             {});
+        checkSplit(",,",            {});
+        checkSplit(" ,",            {});
+        checkSplit(" , ,",          {});
+        checkSplit("x",             {"x"});
+        checkSplit(" x",            {"x"});
+        checkSplit(" \t x",         {"x"});
+        checkSplit("x ",            {"x"});
+        checkSplit("x \t",          {"x"});
+        checkSplit(" \t x \t ",     {"x"});
+        checkSplit("\"\"",          {});
+        checkSplit(" \"\"",         {});
+        checkSplit("\"\" ",         {});
+        checkSplit("\"x\"",         {"x"});
+        checkSplit("\" \"",         {" "});
+        checkSplit("\" x\"",        {" x"});
+        checkSplit("\"x \"",        {"x "});
+        checkSplit("\" x \"",       {" x "});
+        checkSplit("\"\tx \"",      {"\tx "});
+        checkSplit("x,y",           {"x", "y"});
+        checkSplit("x ,\ty ",       {"x", "y"});
+        checkSplit("x, y, z",       {"x","y","z"});
+        checkSplit("x, \"y\", z",   {"x","y","z"});
+        checkSplit(",,x,,\"y\",,",  {"x","y"});
+    }
+
+    void
+    checkIter(std::string const& s,
+        std::vector<std::string> const& expected)
+    {
+        std::vector<std::string> got;
+        std::for_each(list_iterator{&s[0], &s[s.size()]},
+            list_iterator{&s[s.size()], &s[s.size()]},
+                [&](string_view const& v)
+                {
+                    got.emplace_back(v);
+                });
+        expect(got == expected);
+    }
+
+    void
+    testIter()
+    {
+        checkIter("x",              {"x"});
+        checkIter(" x",             {"x"});
+        checkIter("x\t",            {"x"});
+        checkIter("\tx ",           {"x"});
+        checkIter(",x",             {"x"});
+        checkIter("x,",             {"x"});
+        checkIter(",x,",            {"x"});
+        checkIter(" , x\t,\t",      {"x"});
+        checkIter("x,y",            {"x", "y"});
+        checkIter("x, ,y ",         {"x", "y"});
+        checkIter("\"x\"",          {"x"});
```

Comment:
Nice tests. I'd add one for a string with a comma and white space in. Something like: `checkIter("\"x, y z\", a",  {"x, y z", "a"});`